### PR TITLE
The initial version of the api

### PR DESCRIPTION
**Pull request by d1snin**
updated and ready for extensions api (if you can call it that) for bigrat.monster
powered by Kotlin, Ktor and Koin

### CODE REVIEW

**Review by sourTaste000 — CHANGES_REQUESTED** [review]
why are you uploading the gradle wrapper

**Comment by d1snin** [comment]
ci/cd

File: api/src/xyz/d1snin/api/modules/RandomModule.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+private val random get() = (0..255).random()
+
+fun Routing.randomModule() {
+
+    val controller by inject<RandomController>()
+
+    get("/random") {
+        val rawParam = call.request.queryParameters["raw"].toBoolean()
+
+        call.respondHtml {
```
Comment (sourTaste000) [inline api/src/xyz/d1snin/api/modules/RandomModule.kt:20]: do you really need to use kotlinx for this? multiline string literals works too imo

File: api/resources/application.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+ktor {
+    deployment {
+        port = 8080
+        port = ${?PORT}
```
Comment (sourTaste000) [inline api/resources/application.conf:4]: what is this extra port doing here

File: api/src/xyz/d1snin/api/modules/RandomModule.kt (hunk 1 above)
Comment (d1snin) [reply api/src/xyz/d1snin/api/modules/RandomModule.kt:20]: if you are talking about DSL, then yes, it is very convenient and concise in the first place

File: api/resources/application.conf (hunk 2 above)
Comment (d1snin) [reply api/resources/application.conf:4]: this file was generated by idea automatically

File: api/src/xyz/d1snin/api/modules/RandomModule.kt (hunk 1 above)
Comment (d1snin) [reply api/src/xyz/d1snin/api/modules/RandomModule.kt:20]: cope

**Comment by EmeraldSnorlax** [comment]
[please move this pull request](https://github.com/bigratmonster/api-kt)

**Pull request closed without merging**